### DISCUSSION TITLE
Delegating ActiveRecord update and destroy methods to Tramway Decorator by default

### DISCUSSION THREAD

**Discussion by kalashnikovisme** [Purple-Magic/tramway#46] [General]
Wanna implement delegation of 2 AR methods to Tramway Decorator. So, we will be able to do something like this

What must we do now?

```ruby
user = tramway_decorate User.first
user.object.destroy
```

After implementation it would be

```ruby
user = tramway_decorate User.first
user.object.destroy
```

What do you think?

**Comment by kalashnikovisme** [comment]
@haukot @moshinaan @tkarpesh WDYT? :slightly_smiling_face:

**Comment by kalashnikovisme** [comment]
Draper does not delegate these methods by default.

**Reply by haukot** [reply]
Seems legit, shouldn't this be in Tramway Forms then?

**Reply by kalashnikovisme** [reply]
In this case, decorators will look like this

```ruby
class UserDecorator < Tramway::BaseDecorator
  delegate_attributes :email, :role

  delegate :update, to: :object
  delegate :destroy, to: :object
end
```

1. It would be better to describe only decoration in decorators.
2. Maybe, let's do a helper `behave_as_ar` or something like that. This method will delegate `update` and `destroy` methods.

**Reply by haukot** [reply]
I see. Then yes, it seems it's better to create a helper method. 
As 
1. Adding :update and :destroy method by default contradicts the current way where an user should explicitly list all delegations
2. Decorators may work not only with AR models
3. There are more useful AR methods which users could want, e.g. `.validate`, `.save`, `.with_lock`, etc

**Reply by haukot** [reply]
Btw `delegate_all` also looks like a promising helper method :)

**Reply by kalashnikovisme** [reply]
What do you mean `promising helper method`? :slightly_smiling_face:

**Reply by haukot** [reply]
I mean it also could be useful

**Reply by kalashnikovisme** [reply]
What about `delegate_all`, I think that this gun is too big :slightly_smiling_face:

**Comment by kalashnikovisme** [comment]
[Release 0.4.3](https://github.com/Purple-Magic/tramway/releases/tag/0.4.3) contains updates.
